### DISCUSSION TITLE
请问 1.8.3 版本使用内置DNS时新增加的默认策略是非 A AAAA 的查询丢弃，是出于什么考虑？

### DISCUSSION THREAD

**Discussion by chika0801** [XTLS/Xray-core#2393] [Q&A]
我在自己的一些VPS上，通过设置DNS地址为127.0.0.1，把DNS查询流量导入Xray来测试，目前使用了10天左右，没有发现日常客户端使用有异常。

请问把默认值动作（非 A AAAA 的查询丢弃）这样设计是出于哪方面考虑，可能简单介绍一下吗？

以下是GPT的回答
```
ChatGPT
自建DNS并只设置解析 A 和 AAAA 记录，同时屏蔽其他查询类型可能会导致一些问题和限制，其中一些主要坏处包括：

缺少功能性：屏蔽其他查询类型意味着该DNS服务器将无法处理其他重要的查询，如CNAME、MX、TXT等记录类型。这可能会导致邮件发送和接收问题、域名验证问题以及其他与其他记录类型相关的功能缺失。

安全问题：虽然屏蔽一些查询类型可能会限制一些潜在的攻击渠道，但如果配置不当，也可能会增加DNS服务器的易受攻击性。例如，如果某些查询类型被屏蔽而没有适当的安全措施，可能会使服务器容易受到DNS欺骗（DNS spoofing）等攻击。

互联网不兼容性：DNS是互联网上重要的基础设施之一，各种设备和服务都依赖于不同的DNS查询类型。如果你的自建DNS不支持这些查询类型，那么这些设备和服务可能无法正常工作。

域名管理复杂性：如果只提供A和AAAA记录解析，其他记录类型的配置可能需要使用其他DNS服务提供商或者其他方式，导致域名管理变得更加复杂。

降低可用性：限制DNS功能可能会降低域名的可用性和可访问性。例如，如果邮件服务器无法解析MX记录，邮件可能无法正确地发送到该域名。

综上所述，只设置A和AAAA记录并屏蔽其他查询类型可能会导致功能受限、安全性问题和互联网不兼容性。如果没有特殊原因要求，建议使用完整功能的DNS服务提供商，以确保域名和网络服务的正常运作。
```

**Comment by chika0801** [comment]
~想到一个远古帖子提到的是和IOS上用NF APP 看片有关，我想这样设计肯定不是解决看NF IP掉全解锁的问题~

https://hostloc.com/thread-1053477-1-1.html

**Comment by RPRX** [comment]
https://github.com/XTLS/Xray-core/discussions/2211#discussioncomment-6183390
https://github.com/XTLS/Xray-core/pull/2131#issuecomment-1596292941

简单来说原来的行为是不应该的，183 先改成了默认丢弃，有人遇到问题再继续改

**Comment by nursery01** [comment]
其實gpt回答具有深度的it問題只有兩種結果，第一種是答錯，第二種是回答的答案跟我的問題有關嗎？

我叫它給我寫段代碼，直接給我寫了個”死亡漩渦”丟給我，企圖耗盡我的CPU，享受惡作劇後的愉悅
